applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -533,10 +533,20 @@ static void stream_stopped_cb(struct bt_bap_stream *stream, uint8_t reason)
 		headsets[channel_index].hci_wrn_printed = false;
 	}
 
-	if (!ep_state_check(headsets[AUDIO_CH_L].sink_stream.ep, BT_BAP_EP_STATE_STREAMING) &&
-	    !ep_state_check(headsets[AUDIO_CH_R].sink_stream.ep, BT_BAP_EP_STATE_STREAMING)) {
-		ret = ctrl_events_le_audio_event_send(LE_AUDIO_EVT_NOT_STREAMING);
-		ERR_CHK(ret);
+	if (stream == &headsets[AUDIO_CH_L].sink_stream) {
```

Comment:
Would this mean if one of headset disconnected, then gateway will stop audio streaming?

Reply:
No, this checks if the stream that wasn't stopped is still in a streaming state, and if so, don't stop the audio. The problem with the old code was that the ep hadn't changed from streaming to not streaming yet in this callback, so we never stopped the audio.